.travis.yml

```diff
@@ -102,36 +102,63 @@ geosx_pangea_build: &geosx_pangea_build
   - gsutil cp -a public-read ${GEOSX_BUNDLE} gs://geosx/Pangea2/
 
 stages:
+  - check_code_style_and_docs
   - build
   - check_submodules
 
 jobs:
   include:
-  - <<: *geosx_linux_build
+  - stage: check_code_style_and_docs
+    name: code_style
+    <<: *geosx_linux_build
+    env:
+    - DOCKER_REPOSITORY=geosx/centos7.7.1908-clang9.0.0
```

Comment:
`ubuntu18.04-gcc8` weights ~380MB while `centos7.7.1908-clang9.0.0` is about 2.1 GB.
Since it should be equivalent, I would pick the lightest image.

Reply:
@TotoGaz @bmhan12 
When we replace `uncrustify` with `clang-format`, we will need a lightweight image with clang. Any idea how heavy that image will be?

Reply:
We could see that changing the docker image did save 3 minutes on a 5 minutes stage: the information comes as fast as possible.
The work to do this was minimal: just changing the name in the `.travis.yml`.

For the future steps, do we provide `clang-format` in our TPL yet? I'm not sure but I'd be surprised.
I could see some options
- Add `clang-format` in our `clang` docker images (not TPL). I'd be surprised that it would weight so much more. Then we could use them for code formatting check. It would not be light, so we would not save those 3 minutes, but we have to balance with other solutions that would require more work.
- Build our own docker images with just `clang-format` and what we need in it. That would require a side build of docker image. FWIW I've already build `clang` from scratch so that would just mean adapting some scripts. That may be interesting if we want to use our own `clang` and not the `rpm` provided by LLNL too.
- Last option that could be very efficient if we manage to get it work, would be to use the ubuntu OS provided by `travis-ci` without any docker layer. _E.g._ ubuntu focal 20.04 provides https://packages.ubuntu.com/focal/clang-format-10 (and I think that they keep backporting it to ubuntu focal so we could stay up to date if needed). But maybe do we need `ctest` or additional packages. Are they available on ubuntu focal with compatible versions? To be investigated.

If we manage to make it work, I would vote for option 3 since it would also mean minimal maintenance on our sides.